ci/terraform/modules/endpoint-lambda/variables.tf

```
@@ -149,3 +149,13 @@ variable "dynatrace_secret" {
   })
   sensitive = true
 }
+
+variable "snapstart" {
+  type    = bool
+  default = false
+}
```

I wonder if it would be clearer to make the exclusivity explicit, rather than the changes to concurrency config you've made above?
```suggestion
variable "snapstart" {
  type    = bool
  default = false
  validation {
    condition = var.provisioned_concurrency == 0 || !var.snapstart
    error_message = "Snapstart and provisioned concurrency cannot be used together"
  }
}
```

---

issue: can't refer to other variables within the condition